packages/eslint-plugin/src/util/misc.ts

```diff
@@ -225,8 +225,8 @@ function isRestParameterDeclaration(decl: ts.Declaration): boolean {
   return ts.isParameter(decl) && decl.dotDotDotToken != null;
 }
 
-function isParenlessArrowFunction(
-  node: TSESTree.ArrowFunctionExpression,
+function isParenlessFunctionExpression(
```

Comment:
For me, `isParenlessFunctionExpression` evokes this:
```ts
(function iifeWithParens() {})();
// vs
+function iifeNoParens() {}();

// or

arr.map(function (x) { return x; });
// vs
arr.map((function (x) { return x; }));
```

rather than

```ts
(x) => y;
// vs
x => y;
```

Let's do one or multiple of
- Keep the name specific to arrow functions
- Keep the `node` type unchanged and just check the type of the node in SBE before calling it (I would definitely recommend this, since otherwise we should probably short-circuit false if the node is not an arrow function expression in the implementation, at which point... why even let it be called on non- arrow functions)
- Add JSDoc to be more clear what this function is even checking for.

Reply:
I see your point, it's definitely confusing.

It's also completely redundant (like you mentioned). I've reverted the changes to `isParenlessArrowFunction` and added a check verifying the node is an arrow function expression.